src/funcx_common/task_storage/s3.py

```diff
@@ -56,18 +56,40 @@ def _store_to_s3(self, task: TaskProtocol, result: str) -> None:
             }
 
     def _get_from_s3(self, task: TaskProtocol) -> str:
-        assert task.result_reference
+        result_ref = task.result_reference
+        if result_ref is None:  # pragma: no cover
+            raise StorageException(
+                f"task {task.task_id} result reference was None inside of _get_from_s3"
+            )
         try:
-            response = self.client.get_object(
-                Bucket=task.result_reference["s3bucket"],
-                Key=task.result_reference["key"],
+            bucket = result_ref["s3bucket"]
+            key = result_ref["key"]
+        except KeyError as err:
+            raise StorageException(
+                "task {task.task_id} result_reference pointed to S3, but was missing "
+                "s3bucket or key"
+            ) from err
+
+        if not isinstance(bucket, str):
```

Comment:
I don't think these type-checks add much since this is the class that is writing the result_ref.

Reply:
I don't see any justification for _not_ doing them.

Any code in the funcx services could modify a task and write that field. Especially if new versions of the storage class roll out with new behaviors, being pedantic about everything guards against unexpected misbehavior.